proto/frequenz/api/common/v1/pagination.proto

```diff
@@ -0,0 +1,43 @@
+// Parameters for pagination.
+//
+// Copyright 2023 Frequenz Energy-as-a-Service GmbH
+//
+// Licensed under the MIT License (the "License");
+// you may not use this file except in compliance with the License.
+
+syntax = "proto3";
+
+package frequenz.api.common.v1.pagination;
+
+// A message defining parameters for paginating list requests.
+// It can be appended to a request message to specify the desired page of
+// results and the maximum number of results per page. For initial requests
+// (requesting the first page), the page_token should not be provided. For
+// subsequent requests (requesting any page after the first), the
+// next_page_token from the previous responses PaginationInfo must be supplied.
+// The page_size should only be specified in the initial request and will be
+// disregarded in subsequent requests.
+message PaginationParams {
+  // Optional. The maximum number of results to be returned per request.
+  uint32 page_size = 1;
+
+  // Optional. The token identifying a specific page of the list results.
+  string page_token = 2;
+}
+
+// A message providing metadata about paginated list results.
+// The PaginationInfo message delivers metadata concerning the paginated list
+// results and should be appended to the response message of a list request. The
+// total_items field must be set to the total count of items that adhere to the
+// filter criteria defined in the request. The next_page_token field should be
+// populated with the token to be used in the subsequent request to fetch the
+// next page of results. If there are no additional results, the next_page_token
+// field should be omitted.
```

Comment:
> If there are no additional results, the next_page_token field should be omitted.

Should the `next_page_token` be optional then?

Reply:
Yes, but should we add it since it is in the response message and is handled by our API implementation?

Reply:
Yes, we should. Otherwise the implementations cannot return `None` when there is no next page.

Reply:
> Should the next_page_token be optional then?

no because this is used in a messages response and being set by the server. it can just be that this is empty. I mean we can add the keyword optional if that also applies to things that are part of the servers response message.

Reply:
Isn't every field optional in v3 of protobuf or why should this not be possible then?

Reply:
I think `None` is semantically better than an empty string to represent NULL. So I'd vote for an optional type in such cases.

Reply:
>Isn't every field optional in v3 of protobuf or why should this not be possible then?

No, scalar types are not optional by default.

Reply:
fine to follow @tiyash-basu-frequenz lead here. but lets make this clear in the documentation.

Reply:
Ok, so i should add the prefix to the Pagination Params as well? 
Like this

```
...
optional page_size
...
optional page_token
...
optional next_page_token
...
```

Reply:
`page_size` and `page_token` should not be optional IMO, because these won't be null. Only the `next_page_token` field should be `optional` because it can be null.

Reply:
I am not 100% sure if i understand this correctly, so to clarify:
They should not be optional because they either have a value, for fetching the next request, or are omitted for the initial request. But specifically setting it to null doesn't mean anyhting in that sense, so it shouldnt be optional.
But if next_page_token is optional and omitted, it is set to null and this specifically conveys that this is the last response for the reqeust?

Reply:
When a service returns this object, then the following things will happen:

a. A page is being returned, therefore it will have a size. So `page_size` will always contains the number of items in the returned page.
b. A page is being returned, so it must have a token. So `page_token` will always representt the token of the returned page, which will never be null.
c. There may or may not be a next page. Therefore `next_page_token` is an optional field.

Reply:
`page_size` and `page_token` are only for the request message. So if the initial request is sent, the `page_token` should be omitted to signal this is the initial request. 
`page_size` is communicated in the initial request, so unless we want the user to have the ability to adjust the `page_size` of every response this only needs to be specified in the initial request.

Reply:
> so unless we want the user to have the ability to adjust the page_size of every response this only needs to be specified in the initial request

I think it makes sense to offer that ability. The standard for most APIs is that the page size can be adjusted on subsequent requests.

Reply:
Here is an example how it usually works with REST:

Initial request: GET /items?page_size=10
Response includes next_page_token="xyz"
Subsequent request: GET /items?page_size=20&next_page_token=xyz